app/src/main/java/com/pajato/android/gamechat/chat/fragment/SelectForInviteFragment.java

```diff
@@ -161,56 +203,19 @@ private void processSelection(@NonNull final ClickEvent event, @NonNull final Ch
         // If the item is a group, then if it's selected, also select it's common room. If it's
         // deselected, deselect all of it's rooms.
         if (clickedItem.type == inviteGroup) {
```

Comment:
I have a hunch that your algorithm might benefit from just paying attention to the selected items when and only when the INVITE button is clicked.